pkg/kubefed2/federate/federate.go

```diff
@@ -190,44 +197,50 @@ func getTargetResource(hostConfig *rest.Config, typeConfig *fedv1a1.FederatedTyp
 	return resource, nil
 }
 
-func createFedResource(hostConfig *rest.Config, typeConfig *fedv1a1.FederatedTypeConfig, template *unstructured.Unstructured, dryrun bool) (*unstructured.Unstructured, error) {
+func FederatedResourceFromTargetResource(typeConfig typeconfig.Interface, targetResource *unstructured.Unstructured) (*unstructured.Unstructured, error) {
 	fedAPIResource := typeConfig.GetFederatedType()
-	fedClient, err := ctlutil.NewResourceClient(hostConfig, &fedAPIResource)
-	if err != nil {
-		return nil, errors.Wrapf(err, "Error creating client for %s", fedAPIResource.Kind)
-	}
 
-	targetKind := typeConfig.GetTarget().Kind
-	if targetKind == ctlutil.ServiceAccountKind {
-		unstructured.RemoveNestedField(template.Object, ctlutil.SecretsField)
+	if typeConfig.GetTarget().Kind == ctlutil.ServiceAccountKind {
```

Comment:
Is there any negative consequence to federating services with their controller-set fields (e.g. clusterip, ports)?

Reply:
I think it makes sense to clear these fields also; Both cluster IP and nodeport can be user set also (similar to secret in serviceaccount). User would need to update the federated resource explicitly, if that is what he intends to do.